src/main/java/frc/robot/subsystem/ElevatorSubsystem.java

```diff
@@ -0,0 +1,366 @@
+            elevatorPower = 0;
+        }
+        if (wristPower > 0 && wristPos.gt(MoPrefs.wristMaxExtension.get())) {
+            wristPower = 0;
+        }
+        if (wristPower < 0 && wristPos.lt(Units.Rotations.zero())) {
+            wristPower = 0;
+        }
+
+        return new ElevatorMovementRequest(elevatorPower, wristPower);
+    }
+
+    public ElevatorPosition getElevatorPosition() {
+        return new ElevatorPosition(elevatorRelEncoder.getPosition(), wristRelEncoder.getPosition());
+    }
+
+    public void adjustDirectPower(ElevatorMovementRequest request) {
+        request = limitElevatorMovementRequest(request);
+        elevatorA.set(request.elevatorPower);
+        elevatorWrist.set(request.wristPower);
+    }
+
+    public void adjustVelocity(ElevatorMovementRequest request) {
+        request = limitElevatorMovementRequest(request);
+
+        Measure<LinearVelocityUnit> elevatorVelocity =
+                MoPrefs.elevatorMaxRps.get().times(request.elevatorPower);
+        Measure<AngularVelocityUnit> wristVelocity = MoPrefs.wristMaxRps.get().times(request.wristPower);
+
+        elevatorVelocityPid.setVelocityReference(elevatorVelocity);
+        wristVelocityPid.setVelocityReference(wristVelocity);
+    }
+
+    public void adjustElevatorSmartPosition(ElevatorPosition position) {
+        elevatorSmartMotionPid.setPositionReference(position.elevatorDistance);
+    }
+
+    public void adjustWristSmartPosition(ElevatorPosition position) {
+        wristSmartMotionPid.setPositionReference(position.wristAngle);
+    }
+
+    public void adjustSmartPosition(ElevatorPosition position) {
+        switch (state) {
+            case NORMAL:
+            default:
+                if (position.equals(ElevatorSetpointManager.getInstance().getSetpoint(ElevatorSetpoint.INTAKE))
+                        && atPosition(position)) {
+                    disableWristReverseLimit();
+                    new ScheduleCommand(tiltback);
+                    state = WristState.HOLDING;
+                } else if (!atHeight(position) && !stowedWrist(position)) stowWrist();
+                else if (stowedWrist(position)) adjustElevatorSmartPosition(position);
+                else adjustWristSmartPosition(position);
+                break;
+            case HOLDING:
+                if (!tiltback.isScheduled()) holdWristIn();
+                if (!atHeight(position)) {
+                    if (tiltback.isScheduled()) tiltback.cancel();
+                    state = WristState.RETURNING;
+                }
+                break;
+            case RETURNING:
+                stowWrist();
+                if (stowedWrist(position)) {
+                    enableWristReverseLimit();
+                    state = WristState.NORMAL;
+                }
+                break;
+        }
+    }
+
+    public void stowWrist() {
+        wristSmartMotionPid.setPositionReference(Units.Rotations.zero());
```

Comment:
The stow position should be retrieved from the `STOW` setpoint. We can't assume it's zero.